projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -443,34 +443,94 @@ function wpcom_launchpad_track_completed_task( $task_id, $extra_props = array()
  * Mark a task as complete.
  *
- * @param string $task_id The task ID.
+ * @param string|string[] $task_ids The task IDs that should be marked as complete.
  * @return bool True if the task was marked as complete, false otherwise.
  */
-function wpcom_mark_launchpad_task_complete( $task_id ) {
+function wpcom_mark_launchpad_task_complete( $task_ids ) {
+	$task_ids = is_array( $task_ids ) ? $task_ids : array( $task_ids );
+	$values   = array_fill_keys( $task_ids, true );
+
+	$result = wpcom_launchpad_update_task_status( $values );
+
+	return $result;
+}
+
+/**
+ * Mark a task as incomplete. This method is here for backwards compatibility of
+ * the Rest API that allowed marking a task as incomplete.
+ *
+ * @param string|string[] $task_ids The task IDs that should be marked as complete.
+ * @return bool True if the task was marked as complete, false otherwise.
+ */
+function wpcom_mark_launchpad_task_incomplete( $task_ids ) {
+	$task_ids = is_array( $task_ids ) ? $task_ids : array( $task_ids );
+	$values   = array_fill_keys( $task_ids, false );
+
+	return wpcom_launchpad_update_task_status( $values );
+}
+
+/**
+ * Updates task/s statuses. If a non-existent task is passed, it will be ignored.
+ * It will trigger the completion event in Tracks for the tasks that were marked as complete.
+ *
+ * @param array $new_statuses Array of mappings [ task_id: string => status: bool ].
+ * @return bool True if the option where the statuses are stored was updated, false otherwise.
+ */
+function wpcom_launchpad_update_task_status( $new_statuses ) {
 	$task_definitions = wpcom_launchpad_get_task_definitions();
+	$reverse_id_map   = wpcom_launchpad_get_reverse_id_mappings();
 
-	// If the task ID isn't defined, return false.
-	if ( ! isset( $task_definitions[ $task_id ] ) ) {
-		return false;
-	}
+	$statuses               = array();
+	$original_status_values = (array) get_option( 'launchpad_checklist_tasks_statuses', array() );
 
-	// If the task has an id_map, use that instead.
-	$key = $task_id;
-	if ( isset( $task_definitions[ $task_id ]['id_map'] ) ) {
-		$key = $task_definitions[ $task_id ]['id_map'];
-	}
+	foreach ( $new_statuses as $task_id => $new_value ) {
+		// Find out wether the task id sent is the actual task id or the id_map.
+		$actual_task_id = ( isset( $reverse_id_map[ $task_id ] ) ) ? $reverse_id_map[ $task_id ] : $task_id;
+		if ( ! isset( $task_definitions[ $actual_task_id ] ) ) {
+			// Bail if no task definition was found.
+			continue;
+		}
 
-	$statuses         = get_option( 'launchpad_checklist_tasks_statuses', array() );
-	$statuses[ $key ] = true;
-	$result           = update_option( 'launchpad_checklist_tasks_statuses', $statuses );
+		// If the task has an id_map, use that, otherwise use the actual task id as the key.
+		$task                        = $task_definitions[ $actual_task_id ];
+		$stored_task_id              = isset( $task['id_map'] ) ? $task['id_map'] : $actual_task_id;
+		$statuses[ $stored_task_id ] = $new_value;
```

Comment:
Should we ensure that we're only storing booleans?
```suggestion
		$statuses[ $stored_task_id ] = (bool) $new_value;
```

Reply:
yes, nothing else should be stored there.